compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -2160,6 +2160,9 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                 this.lower_param_bounds_mut(&param.bounds, itctx.reborrow()).collect()
             });
 
+        // lower here so const-defaults can be lowered
+        let hir_id = self.allocate_hir_id_counter(param.id);
```

Comment:
Why did you need to do this? (Sorry if I asked this before, I'm trying to remind myself.)

Reply:
Hm, I can't really remember myself, but looking at the code below, I think it was necessary to have this hir already allocated so that the anon consts below have a parent I believe.

Reply:
This seems like the most likely cause of the problems to me. I don't suppose you could avoiding this and lowering normally to see what happens? (It may be something like this has to be done, but needs to be tweaked slightly.)

Reply:
do you mean just remove that specific line and see if it works?

Reply:
I meant not lowering the node there, and just doing `default.as_ref().map(|def| this.lower_anon_const(def))` without `with_hir_id_owner` for the anon const.

Reply:
This seems much better now @JulianKnodt, no? It seems there's just a failing test.

Reply:
ah certainly, not sure why I chose to do it like that in the past

Reply:
Let me know if it seems like you can make progress from here, or if you hit a dead end again!

Reply:
ah yea yea my bad just been caught up in another project recently

Reply:
No rush!: I haven't been able to help for a while either, so I know what that's like :)